modules/asset-manager-storage-fs/src/main/java/org/opencastproject/assetmanager/storage/impl/fs/AbstractFileSystemAssetStore.java

```diff
@@ -176,7 +176,11 @@ private void mkParent(File f) {
 
   /** Create this directory and all of its parents. */
   protected void mkDirs(File d) {
-    if (d != null && !d.exists() && !d.mkdirs()) {
+    if (d == null) {
```

Comment:
Why not 

`if (d != null && !d.mkdirs() && !d.exists()) {`

?

Reply:
Because that was my first change and the reaction to that was that it's hard to understand.
I hoped that removing the somewhat unrelated null check would help a bit while we don't lose anything except for two lines ;-)

Reply:
... and the somewhat weird use of the if statement.

Anyway, while I personally don't like if statements the jump out of the method, I don't have a strong opinion about that, so fine for me.